RELEASE.md

```diff
@@ -0,0 +1,101 @@
+  - Label the PR: `status: cherry-picked 🍒`.
+
+_Outcome_: __Release branch has all relevant changes merged & pushed.__
+
+### Prepare release
+#### Ensure release branch readme is up to date
+- Run changelog script `npm run changelog` to get changelog txt for readme. Changelog content will be output to screen by script.
+- Add changelog section for release, e.g. `= 2.5.11 - 2020-01-20 =`.
+- Copy-paste the changelog content into `readme.txt`.
+- Make any other changes to readme as needed - e.g. support versions changing, new blocks. 
+- Push readme changes to release branch on origin repo.
+
+_Outcome_: __`readme.txt` is up to date with changes for release.__
+
+#### Build zip & smoke test
+- Ensure you are on tip of release branch.
+- Update dependencies – `npm ci`.
+- Run a production build - `npm run build`.
+- Run package script to get a zip to test `npm run package-plugin`.
+- Smoke test built release zip:
+  - Recommend doing this in a clean environment, e.g. Jurassic.Ninja site.
+  - Sanity check - check correct blocks exist and function.
+  - Test to confirm new features/fixes are working correctly.
+  - Smoke test – test a cross section of core functionality.
+
+_Outcome_: __Confident that source code is ready for release: intended fixes are working correctly, no release blockers or build issues.__
+
+### Release!
+#### Release to GitHub
+- Prepare tagged release on github `npm run deploy`. 
+  - Note: the script automatically updates version numbers (commits on your behalf).
+- Edit release, add changelog info to Github release notes.
+- Check release repo tag is correct - checkout, smoke test/sanity check.
+
+_Outcomes_: __Version numbers updated in source code & developers can test tagged release.__
+
+##### Release to WPORG
+- Push release to WPORG `npm run release`.
+- Note that this may happen a day or so after Github release.
+- FYI: Sometimes we make small changes (e.g. readme only) by [committing directly to svn repo](https://developer.wordpress.org/plugins/wordpress-org/how-to-use-subversion/#editing-existing-files).
+
+_Outcome_: __Customers can install/update via WPORG; WPORG plugin page is up to date__.
+
+### After release
+#### Update `master` with release changes
+- Ensure changelog is up to date on master.
+- If major/minor release, update version on master with dev suffix, e.g. [`2.6-dev`](https://github.com/woocommerce/woocommerce-gutenberg-products-block/commit/e27f053e7be0bf7c1d376f5bdb9d9999190ce158).
+- Code changes should already be present on master - all dev happens on `master`.
+
+
+## `@todo` Missing info to link to / document
+- Release milestones
+- Relationship to Woo core release
+- Testing blocks plugin as included in Woo core (do we do this?)
+- Policy for inclusion in major / minor / fix releases - i.e. how we use versioning 
+- ["Legacy"](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/add/release-process/assets/js/legacy/README.md) stuff - is this relevant to releases?
```

Comment:
Another item that came up recently is around ensuring all Woo core packages ( Blocks, WC-Admin, REST API ) are running the same version of the jetpack-autoloader. So that might be another good one to add to the todo list here... thought we have yet to determine the workflow around when that should be updated across all 3 repos.

Reply:
> Another item that came up recently is around ensuring all Woo core packages ( Blocks, WC-Admin, REST API ) are running the same version of the jetpack-autoloader.

Great to bring this up, thanks! However, this will be tricky to consider as a part of the release process because you could get a specific release in alignment version wise but then either of the other repositories could bump jetpack autoloader version before the release is pulled in at some later time (at which point the problem will surface anyways).  This is another reason why I think we have to switch to ranges for jetpack autoloader version instead of pinning to a specific version.  I think we should leave that out of this iteration of the release notes until we've resolved how this will work (and hopefully won't be something needing thought about in a release).

Reply:
🐲 

I've added mention of this to that section. Note that this is not a list of things that the release process should document - it's a list of things that might be related and might need clarification/documenting somewhere.

Reply:
> I've added mention of this to that section. Note that this is not a list of things that the release process should document - it's a list of things that might be related and might need clarification/documenting somewhere.

I appreciate having this list but I don't see the need to keep it in the release documentation since it's not relevant to someone following instructions for doing a release.  Some of these items are also listed as questions which can be answered without documenting.  If you see any of these as important things that we should handle, then create an issue for it.  However, to help with deciding whether something needs an issue (or faq documentation somewhere), I've gone through and addressed the items below 👇 

**Release milestones**

I don't know what you mean by this item.  Do you mean documenting when release milestones get created? Do you mean how release milestones differ from Zenhub releases?

**Relationship to Woo core release**

This is another bullet point that seems vague for a todo.  I think we already have some info on this in the project's README.md: 

> Feature plugin for WooCommerce + Gutenberg. This plugin serves as a space to iterate and explore new Blocks for WooCommerce, and how WooCommerce might work with the block editor.
>
> If you're just looking to use these blocks, update to the latest version of WooCommerce! The blocks are bundled into WooCommerce since version 3.6, and can be added from the "WooCommerce" section in the block inserter.

**Testing blocks plugin as included in Woo core (do we do this?)**

If you mean our squad, we do and we don't.  We do, because what is in woo core is exactly what is in the tagged release (because the tag is pulled in via composer).  So essentially, _yes, it's been tested_. We don't, because we don't explicitly test the pull request that bumps the woo blocks version in composer.json for Woo Core.  I don't think it's necessary for our squad to do because when our plugin is bumped in Woo Core, there is testing done before Woo core is published.  Any issues with blocks will either be bugs we didn't catch in our own testing (i.e. not specifically from including in woo core) or because something went wrong with the woo core release process not pulling in the woo blocks package correctly (which is not something specific to the plugin itself).

**Policy for inclusion in major / minor / fix releases - i.e. how we use versioning**

I don't think we should be too rigid here when it comes to policy, but it looks like we pretty much follow semver for the most part.

- patch: important fixes that we earmark for including in a minor version release of WC core.  These might also go on wp.org if the minor version on wp.org is the same branch as the patch.
- minor: new features and bug fixes released for the plugin are typically minors. Minor's will get bundelled in whatever version of WC core they are earmarked for.
- major: breaking changes. These would be included in a version of WC core that is a major release (breaking release).

**["Legacy"](https://github.com/woocommerce/woocommerce-gutenberg-products-block/blob/add/release-process/assets/js/legacy/README.md) stuff - is this relevant to releases?**

You never addressed [my comment here](https://github.com/woocommerce/woocommerce-gutenberg-products-block/pull/1614/files#r368958263) which I think is relevant for testing, but when it comes to the question "is this relevant to releases?" the answer is other than testing, no.  I also don't want to invest too heavily in documenting this (besides the [documentation we already have](https://github.com/woocommerce/woocommerce-gutenberg-products-block/tree/master/assets/js/legacy)) until we've had a chance to see this system in use for a while.

**`jetpack-autoloader` versions in this plugin vs. wc-admin vs. WooCommerce core**

I don't think this should be a todo in a "how to release document" for the reasons I've already given. Instead I think it should just be an issue to deal with (see #1625).

Reply:
Thanks for clarifying those points @nerrad - lots of useful info in this comment. I agree this is not the right doc for this content; it all came up for me as things that might be useful in our team docs somewhere. Removing this section now :)

As for what next with this info – let's follow that up in team channel. I'm hesitant to add docs issues for every question I have :)